File: Aula_23_03/Aluno.cs
public class Aluno {
    public string Nome { get; private set; }
    public string Email { get; private set; }

    public Aluno(string nome) {
        Nome = nome;
        Email = GerarEmail(nome);
    }

    private string GerarEmail(string nome) {
        string[] dadosNome = nome.Split(' ');
        return dadosNome[0].ToLower() + "." + dadosNome[dadosNome.Length - 1].ToLower() + "@email.com";
    }
}

File: Aula_23_03/Aluno.cpp
#include <string>
#include <sstream>
#include <vector>
#include <algorithm>

using namespace std;

class Aluno {
private:
    string nome;
    string email;

    string gerarEmail(const string& nome) {
        vector<string> dadosNome;
        stringstream ss(nome);
        string token;
        while (ss >> token) dadosNome.push_back(token);

        string primeiro = dadosNome.front();
        string ultimo = dadosNome.back();
        transform(primeiro.begin(), primeiro.end(), primeiro.begin(), ::tolower);
        transform(ultimo.begin(), ultimo.end(), ultimo.begin(), ::tolower);

        return primeiro + "." + ultimo + "@email.com";
    }

public:
    Aluno(const string& nome) : nome(nome), email(gerarEmail(nome)) {}

    string getNome() const { return nome; }
    string getEmail() const { return email; }
};